src/Libraries/RevitNodes/Application/Document.cs

```diff
@@ -62,6 +66,71 @@ public static Document Current
             get { return new Document(DocumentManager.Instance.CurrentDBDocument); }
         }
 
+        /// <summary>
+        /// Purge unused Elements from the model. This node is not able to purge materials and material assets
+        /// </summary>
+        /// <param name="deepPurge">Similar to clicking the purge button multiple times in Revit.</param>
+        /// <returns></returns>
+        public string PurgeUnused(bool deepPurge)
+        {
+            //The internal GUID of the Performance Adviser Rule 
+            const string purgeGuid = "e8c63650-70b7-435a-9010-ec97660c1bda";
+
+            var performanceAdviserRuleId = new List<PerformanceAdviserRuleId>();
+            var performanceAdviserRules = PerformanceAdviser.GetPerformanceAdviser().GetAllRuleIds();
+
+            //Iterating through all PerformanceAdviser rules looking to find that which matches PURGE_GUID
+            for (int i = 0; i < performanceAdviserRules.Count; i++)
+            {
+                if (performanceAdviserRules[i].Guid.ToString() == purgeGuid)
+                {
+                    performanceAdviserRuleId.Add(performanceAdviserRules[i]);
+                    break;
+                }
+            }
+
+            TransactionManager.Instance.EnsureInTransaction(this.InternalDocument);
+            int purgeCount = purgeElements(this.InternalDocument, performanceAdviserRuleId, deepPurge);
+            TransactionManager.Instance.TransactionTaskDone();
+            if (purgeCount == 0)
+            {
+                return Properties.Resources.NoElementsToPurge;
+
+            }
+            return string.Format(Properties.Resources.PurgedElements, purgeCount);
+        }
+
+        private static int purgeElements(Autodesk.Revit.DB.Document document, List<PerformanceAdviserRuleId> performanceAdviserRuleId, bool runRecursively)
```

Comment:
let's
- return a `List<int>` so we return a list of all element Ids what were deleted - this will be more useful than a count
- rename method to have uppercase `PurgeElements`. Even if it's private, methods have uppercase names